### PR TITLE
chore: improve error message for temp dir failure

### PR DESCRIPTION
**Pull request by rgrinberg**
I improve the error messages for temporary directory creation in the following way:

1. If we try to create inside a non existing directory, the directory name that we failed to create will be included in the error.
2. If the suffix or prefix contain path elements (e.g. `/` on unix), the directory will be rejected. This isn't allowed because it will end up pointing to subdirectories in the temp directory which haven't been created yet.

### CODE REVIEW

**Comment by rgrinberg** [comment]
@pmwhite do you want to review this one?

File: otherlibs/stdune/src/temp.ml [hunk 1]
```diff
@@ -44,7 +44,9 @@ let create_temp_dir ?perms path =
   | Created -> Ok ()
   | Already_exists -> Error `Retry
   | Missing_parent_directory ->
-    Code_error.raise "[Temp.create_temp_dir] called in a non-existing directory" []
+    Code_error.raise
+      "[Temp.create_temp_dir] called in a non-existing directory"
```
Comment (pmwhite) [inline otherlibs/stdune/src/temp.ml:48]: While we're at it, perhaps we should also improve the grammar. "Non-existing" should be "nonexistent".

File: otherlibs/stdune/src/temp.ml [hunk 2]
```diff
@@ -89,6 +93,16 @@ let create ?perms what ~prefix ~suffix =
        Perhaps, we should use something like [_build/.temp] instead? *)
     Filename.get_temp_dir_name () |> Path.of_filename_relative_to_initial_cwd
   in
+  if contains_path_sep prefix
+  then
+    Code_error.raise
+      "Temp.create: invalid prefix contains path elements"
```
Comment (pmwhite) [inline otherlibs/stdune/src/temp.ml:99]: The wording is a bit off. The prefix is invalid _because_ it contains path elements, but the error makes it sound like it is an invalid prefix that also contains path elements. I'd suggest "prefix must not contain path elements". Same for the suffix message as well.

**Review by pmwhite — APPROVED** [review]
Approving, with 2 small suggestions. Feel free to merge after you consider them.

**Pull request merged**